x/ccv/provider/keeper/partial_set_security.go

```diff
@@ -140,18 +140,12 @@ func (k Keeper) CapValidatorSet(ctx sdk.Context, chainID string, validators []ty
 		return validators
 	}
 
-	if validatorSetCap, found := k.GetValidatorSetCap(ctx, chainID); found && validatorSetCap != 0 {
+	if validatorSetCap, found := k.GetValidatorSetCap(ctx, chainID); found && validatorSetCap != 0 && int(validatorSetCap) < len(validators) {
```

Comment:
Changed based on "Code Efficiency Improvements and Optimization Opportunities" suggestions.